components/automate-ui/src/app/modules/desktop/desktop-detail/desktop-detail.component.ts

```diff
@@ -103,17 +79,18 @@ export class DesktopDetailComponent implements OnInit, OnDestroy {
     this.downloadDropdownVisible = false;
   }
 
-  addCheckInLabels(checkInHistory: DailyNodeRunsStatus[]): DailyNodeRunsStatus[] {
-    let numWeeks = Math.floor(checkInHistory.length / 7);
-    checkInHistory.forEach((history: DailyNodeRunsStatus, index: number) => {
+  get labeledCheckInHistory() {
+    const buckets = this.checkInHistory.durations.buckets;
+    let numWeeks = Math.floor(buckets.length / 7);
+    return buckets.map((history: DailyNodeRunsStatus, index: number) => {
       const isStartOfWeek = (index % 7 === 0) && numWeeks > 0;
-      const startOfWeekLabelText = numWeeks > 1 ? `${numWeeks} weeks ago` : `${numWeeks} week ago`;
-      const isToday = index === (checkInHistory.length - 1);
+      const startOfWeekLabelText = numWeeks !== 0 ? `${numWeeks} weeks ago` : `${numWeeks} week ago`;
```

Comment:
Pretty sure this should be a 1 not a zero:
```suggestion
      const startOfWeekLabelText = numWeeks !== 1 ? `${numWeeks} weeks ago` : `${numWeeks} week ago`;
```

Reply:
Ha oops! Thanks @msorens